vendor/views/vendor.py

```diff
@@ -131,14 +127,10 @@ def post(self, request, *args, **kwargs):
         return redirect('vendor:cart')      # Redirect to cart on success
 
 
-class RemoveFromCartView(View):
+class RemoveFromCartView(SetSiteToRequestMixin, View):
     
     def post(self, request, *args, **kwargs):
-        site = get_current_site(request)
-        if hasattr(request, 'site'):
-            site = request.site
-
-        offer = Offer.objects.get(site=site, slug=self.kwargs["slug"])
+        offer = Offer.objects.get(site=request.site, slug=self.kwargs["slug"])
```

Comment:
use the new function here to get the site object instead of modifying the request object.

Reply:
changed to: `site=get_site_from_request(request)`